osgar/explore.py

```diff
@@ -0,0 +1,146 @@
+"""
+  Explore - follow a wall/obstacles
+    (use 2D SICK LIDAR only)
+"""
+import math
+from datetime import timedelta
+
+import numpy as np
+
+from osgar.node import Node
+
+
+SAFE_DISTANCE_STOP = 0.5
+SAFE_DISTANCE_GO = SAFE_DISTANCE_STOP + 0.3
+WALL_DISTANCE = 0.7  # m
+DESIRED_SPEED = 0.4  # m/s
+
+
+def min_dist(data):
+    data = np.array(data)
+    mask = (data > 100)
+    if np.any(mask):
+        return np.min(data[mask]) * 0.001
+    return None 
+
+
+def min_dist_arr(data):
+    """Fake segments as in demo.py for Velodyne"""
+    num = len(data)
+    # laser data are anticlockwise -> swap left, right
+    return min_dist(data[num//2:]), min_dist(data[:num//2])
+
+
+def tangent_circle(dist, radius):
```

Comment:
I don't understand this function.

Reply:
Thanks for the description
```
return tangent angle to a circle placed at (dist, 0.0) with radius=radius
```
but I don't think that is what it does. It actually calculates the angle to a point at `dist` in-front and `radius` to the side. Line from (0,0) to (dist, radius) will cut through the circle.

Reply:
Never mind. It is correct, I had it backwards.